components/rename_channel_modal/rename_channel_modal.jsx

```diff
@@ -161,20 +188,39 @@ export class RenameChannelModal extends React.PureComponent {
         }
 
         channel.name = this.state.channelName.trim();
-        if (!channel.name) {
-            state.nameError = formatMessage(holders.required);
-            state.invalid = true;
-        } else if (channel.name.length > Constants.MAX_CHANNELNAME_LENGTH) {
-            state.nameError = formatMessage(holders.maxLength, {maxLength: Constants.MAX_CHANNELNAME_LENGTH});
-            state.invalid = true;
-        } else {
-            const cleanedName = cleanUpUrlable(channel.name);
-            if (cleanedName === channel.name) {
-                state.nameError = '';
-            } else {
-                state.nameError = formatMessage(holders.lowercase);
-                state.invalid = true;
+        const urlErrors = [];
+        const url = channel.name;
+        if (url) {
+            if (url.length < 2) {
+                urlErrors.push(this.formattedError(holders.minLength));
             }
+
+            if (url.charAt(0) === '-' || url.charAt(0) === '_') {
```

Comment:
I didnt use a regex since I just copied the same url validation from the `change_url_modal`.
Do you think it makes sense to extract the validation into a utility function and refactor it for both of them as part of this PR?

Reply:
if you are using the same validation, I'd definitely go for it, that way if at some point one changes we don't want to allow it to happen somewhere else.